### DISCUSSION TITLE
Ingest Query Params of a Protected Route Prior to Redirect

### DISCUSSION THREAD

**Discussion by neilgamb** [expo/router#163] [Q&A] [answered]
Is it possible to intercept the query parameters of a protected URL prior to the redirect.  For instance, given a simple auth file / route structure:

<img width="227" alt="image" src="https://user-images.githubusercontent.com/26474370/208233551-4ac0a5fc-ec0a-4dc6-863e-1768e1d95a35.png">

When navigating to ie: `http://localhost:19000/details?id=foo`, the details screen is never rendered so there is no way to pull out the route param of `id: 'foo'`.  Here is my root level _layout.tsx:

```
import { Layout } from "expo-router";

import useAuthStore from "../state/auth";

export default function RootLayout() {
  const { isAuthenticated } = useAuthStore();

  return (
    <Layout>
      <Layout.Screen name="(home)" redirect={!isAuthenticated} />
      <Layout.Screen name="signin" redirect={isAuthenticated} />
      <Layout.Children />
    </Layout>
  );
}
```

Im am working on a POC for passwordless auth.  Authentication codes are included as query params in the magic links into the app and in order to link to various protected routes in the app, I need to be able to read in those params for the authentication flow.

**Comment by EvanBacon** [comment, marked as answer]
Sorry for missing this, you should be able to access the query params from the parent layout route:

```diff

import { Layout, useSearchParams } from "expo-router";

import useAuthStore from "../state/auth";

export default function RootLayout() {
+  const params = useSearchParams();
  const { isAuthenticated } = useAuthStore();

  return (
    <Layout>
      <Layout.Screen name="(home)" redirect={!isAuthenticated} />
      <Layout.Screen name="signin" redirect={isAuthenticated} />
      <Layout.Children />
    </Layout>
  );
}
```
